src/examples/java/com/amazonaws/crypto/examples/DataKeyCaching/SimpleDataKeyCachingExample.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except
+ * in compliance with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import javax.xml.bind.DatatypeConverter;
+ 
+import com.amazonaws.encryptionsdk.AwsCrypto;
+import com.amazonaws.encryptionsdk.CryptoMaterialsManager;
+import com.amazonaws.encryptionsdk.MasterKeyProvider;
+import com.amazonaws.encryptionsdk.caching.CachingCryptoMaterialsManager;
+import com.amazonaws.encryptionsdk.caching.CryptoMaterialsCache;
+import com.amazonaws.encryptionsdk.caching.LocalCryptoMaterialsCache;
+import com.amazonaws.encryptionsdk.kms.KmsMasterKey;
+import com.amazonaws.encryptionsdk.kms.KmsMasterKeyProvider;
+
+/**
+ * <p>
+ * Encrypts a string using an AWS KMS customer master key (CMK) and data key caching
+ * 
+ * <p>
+ * Arguments:
+ * <ol>
+ * <li>KMS CMK ARN: To find the Amazon Resource Name of your AWS KMS customer master key (CMK), 
+ *     see 'Viewing Keys' at http://docs.aws.amazon.com/kms/latest/developerguide/viewing-keys.html
+ * <li>Max entry age: Maximum time (in seconds) that a cached entry can be used
+ * <li>Cache capacity: Maximum number of entries in the cache
+ * </ol>
+ */
+public class SimpleDataKeyCachingExample {
+    /*
+     * Security thresholds
+     *   Max entry age is required. 
+     *   Max messages (and max bytes) per data key are optional
+     */
+    private static final int MAX_ENTRY_MSGS = 100;
+    
+    public static byte[] encryptWithCaching(String kmsCmkArn, int maxEntryAge, int cacheCapacity) {
+        // Plaintext data to be encrypted
+        byte[] myData = "My plaintext data".getBytes(StandardCharsets.UTF_8);
+
+        // Encryption context
+        final Map<String, String> encryptionContext = Collections.singletonMap("purpose", "test");
+
+        // Create a master key provider
+        MasterKeyProvider<KmsMasterKey> keyProvider = new KmsMasterKeyProvider(kmsCmkArn);
+
+        // Create a cache
+        CryptoMaterialsCache cache = new LocalCryptoMaterialsCache(cacheCapacity);
```

Comment:
This cache is not very helpful, since it's only being used for a single request. This needs to be refactored to keep the caching CMM around across multiple requests.